include/revng/Pipes/PipelineManager.h

```diff
@@ -122,7 +122,9 @@ class PipelineManager {
 
   /// A helper function used to produce all possible targets. It is used for
   /// debug purposes to see if any particular target crashes.
-  llvm::Error printAllPossibleTargets(llvm::raw_ostream &Stream);
+  llvm::Error produceAllPossibleTargets(llvm::raw_ostream &Stream);
+
+  llvm::Error invalidateAllPossibleTargets(llvm::raw_ostream &OS);
```

Comment:
Why "Possible"?

Reply:
because not all targets that one can build can be produced, not sure i understand the question

Reply:
Would dropping the "Possible" from the name induce anyone in thinking this does something it doesn't actually do?

Reply:
i would not know it is all possible or the current ones

Reply:
What would be the point of producing all the produced targets?
Also, invalidating all produced targets or all targets, produces the same result.

Reply:
yeah, but one is used to see the current state of the system, the other is used see if the propagation of targets across contracts is correct.

Reply:
Can you please clarify what you mean?

Reply:
i mean that when one is trying to understand what is going on with a pipeline  it has written, he usually cares about two things, all the targets currently in each container, and all targets that may be ever produced. The first is needed to understand if the pipeline had loaded what you think has loaded, and the second is used to understand if the contract are being evaluated as you think they should be evaluated.

It is similarly useful in the frontend, where one can have a list all possible targets, where those already produced are marked green.

Reply:
i am ok with replacing allPossible with something different like get allProducible  (even if this makes it looks like that it does not contain the already produced ones) or allAchievable

Reply:
what about `getCurrentTargets()` and `getAllTargets()`